### DISCUSSION TITLE
How to loop until status code or response body field has some specific value?

### DISCUSSION THREAD

**Discussion by asaintsever** [AnWeber/httpyac#820] [General]
Hello,

I am using a REST API that requires to poll an endpoint until the response contains a specific value. I did not find information on how to do this in the documentation (the `loop while` may be the solution but I don't know how to express the condition).

Currently, I use an "ugly" `loop for` combined with `sleep` to do my polling as is:

```
# @loop for 6
# @sleep 10000
# @name deployed_model
GET /_plugins/_ml/tasks/{{model_id}} HTTP/2.0
...defaultHeaders

?? status == 200
?? body model_id exists
?? body state == COMPLETED
```

It is far from being bullet proof (either you poll too long or, worse, not enough). Here I want to poll until `state == COMPLETED` in the response body.
Does httpyac support that in some way? Thanks!

**Comment by AnWeber** [comment]
try this

```
@continueLoop=true
###
# @loop while continueLoop
GET https://httpbin.org/anything

{{
  exports.continueLoop= false; // your condition
}}
```

**Reply by asaintsever** [reply]
Thanks it works perfectly!
